### PR TITLE
Fix in case positions sizes or colors are the same for all points.

### PR DESCRIPTION
**Pull request by couet**
Was found here https://root-forum.cern.ch/t/tscatter-drawing-nothing-with-legacy-cocoa/56067

### CODE REVIEW

File: hist/histpainter/src/TGraphPainter.cxx [hunk 1]
```diff
@@ -4365,6 +4365,22 @@ void TGraphPainter::PaintScatter(TScatter *theScatter, Option_t* chopt)
          maxs = TMath::Max(maxs,theSize[i]);
       }
    }
+   if (minx == maxx) {
+      minx = 0.9*theX[0];
+      maxx = 1.1*theX[0];
```
Comment (jalopezg-git) [inline hist/histpainter/src/TGraphPainter.cxx:4370]: The code seems to be duplicated below for `miny`, `maxy`, `theY`, etc.   Perhaps, we could instead define a lambda expression as in
```c++
auto fnFixRange = [](double &min, double &max, double value) {
   // ...
};
```
and call it accordingly for all the cases below?

Also, could you add a comment on why these values need to be adjusted?

Comment (couet) [reply hist/histpainter/src/TGraphPainter.cxx:4370]: That's super simple protection when min==max ... Which in general is not a good idea, and, in that case, produces wrong results... I guess the code is self-explanatory. A lambda will be a bit confusing for no real added value seems to me.

**Review by linev — COMMENTED** [review]
+-0.1 is not perfect solution. If value is more than 1e12, adding 0.1 has no effect

**Comment by couet** [comment]
@linev I do not understand you point. I did not add  +-0.1, I made a % ....

**Comment by linev** [comment]
> @linev I do not understand your point. I did not add +-0.1, I made a % ....

Sorry, it is really %. But then it does not work for 0.

**Comment by couet** [comment]
your are right for 0. I will take this into account

**Review by linev — APPROVED** [review]

**Pull request merged**